doc/content/step_by_step.md

```diff
@@ -95,72 +107,41 @@ $ hippo-opt -i fluid.i --mesh-only
 
-At present, all AuxVariables used for these transfers must
-be constant and monomial.
 
 ```toml
 # fluid.i
 
-[AuxVariables]
+[FoamVariables]
     [fluid_wall_temp]
-        family = MONOMIAL
-        order = CONSTANT
-        initial_condition = 300
-    []
-    [solid_heat_flux]
-        family = MONOMIAL
-        order = CONSTANT
-        initial_condition = 0
+        type = FoamVariableField
+        foam_variable = T
     []
 []
 ```
+Internally, Hippo creates `AuxVariable`s which can use MOOSE's transfer system and so can be transferred between MOOSE apps.
 
-> +Note:+ *To save on memory, you could actually define a single AuxVariable
-> and transfer the heat flux one way and temperature the other in the same
-> variable. However, for clarity, we use separate variables here.*
-
-### Problem
-
-Next we define the 'Problem' block.
-Hippo provides the `FoamProblem` class to run the OpenFOAM case
-provided in the `[Mesh]` block.
+### Imposing boundary conditions on OpenFOAM from MOOSE
 
-`FoamProblem` has a set of parameters that defines which quantities are
-copied to and from OpenFOAM:
-
-- `foam_heat_flux` -
-  the name of the AuxVariable to copy OpenFOAM's heat flux into.
-- `foam_temp` -
-  the name of the AuxVariable to copy OpenFOAM's wall temperature into.
-- `temp` -
-  the name of an AuxVariable whose values will be used to set
-  a Dirichlet boundary condition on the OpenFOAM solve.
-- `heat_flux`
-  the name of an AuxVariable whose values will be used to set
-  a Neumann boundary condition on the OpenFOAM solve.
-
-In our example, we'll need to set `heat_flux` and `foam_temp`.
+Hippo also implements the `FoamBC` system that allows you to impose boundary conditions on OpenFOAM from the Hippo input file. In this case, we wish to impose a heat flux BC on the temperature field. This requires the `FoamFixedGradientBC`, which mirrors OpenFOAM's `fixedGradient` BC type. As in this case, we provide a heat flux, we must also provide the thermal conductivity, which is specified as the `diffusivity_coefficient`. `kappa` is the name of the thermal conductivity variable in OpenFOAM.
 
 ```toml
 # fluid.i
 
-[Problem]
-    type = FoamProblem
-    # Take the heat flux from MOOSE and set it on the OpenFOAM mesh.
-    heat_flux = solid_heat_flux
-    # Take the boundary temperature from OpenFOAM and set it on the MOOSE mesh.
-    foam_temp = fluid_wall_temp
+[FoamBCs]
+    [solid_heat_flux]
+        type = FoamFixedGradientBC
+        foam_variable = T
+        diffusivity_coefficient = kappa
+    []
 []
 ```
 
+Similar to `FoamVariable`, this creates an `AuxVariable` of the same name under the hood, so the transfer system can be used to provide value of the heat flux to imposed on OpenFOAM.
```

Comment:
This sentence could do with a re-work:

> _of the same name_
  
The same name as what?

> _provide value of the heat flux to imposed_

Some mixed-up/typo-ed grammar here.

Suggestion below, but feel free to re-phrase as you like:

```suggestion
Similar to `FoamVariable`, this creates an `AuxVariable` with name `solid_heat_flux` under the hood. 
MOOSE's transfers system can be used to set the values in the `AuxVariable`, that are then imposed on the OpenFOAM solver's boundary.
```